docs/designs/operator.md

```diff
@@ -0,0 +1,201 @@
+# adx-mon Operator Design
+
+## Overview
+
+The `adx-mon-operator` is a Kubernetes operator responsible for managing the lifecycle of an adx-mon cluster, including its core components (collector, ingestor, alerter) and required Azure infrastructure. The operator leverages [Azure Service Operator (ASO)](https://azure.github.io/azure-service-operator/) to declaratively manage Azure resources such as ADX (Azure Data Explorer) clusters, databases, and other dependencies.
+
+The operator aims to provide a simple, production-ready bootstrap experience for adx-mon clusters, while supporting advanced customization for complex and federated deployments.
+
+---
+
+## Responsibilities
+
+- **Azure Infrastructure Management:**  
```

Comment:
I am curious how managing azure infra using operators pans out because there will be two source of truths - one state in kubernetes and one in azure. The straight greenfield case of creating azure resources from scratch works fine in my head right now but the day 2 operations and reconciliation complications when state changes outside of operator, corner cases where azure apis return errors makes things hairy imo. Just to clarify, I am not objecting. I am actually open to trying this out knowing that we can always end up going back to a bootstrap script just for the infrastructure part.